### PR TITLE
Device Access Validator 

### PR DESCRIPTION
**Pull request by joe94**
### Changes introduced in this PR:
- JWT Validation for device registration is done through bascule now
- Device Access Component

### TODO:
- [x] `contains` (a must for partnerIDs), `intersect` and `equals` might be enough for this first PR. Let me know if that's not the case.

- [x] Update metrics accordingly 
- [x] Unit test updates

### CODE REVIEW

**Comment by joe94** [comment]
fixes #106

File: deviceAccess.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	//Inversed should be set to true if Op should be applied from
+	//valueAt(DeviceCredentialPath) to valueAt(UserCredentialPath)
+	//(Optional)
+	Inversed bool
+}
+
+//deviceAccessCheckConfig drives the device access bussiness logic
+type deviceAccessCheckConfig struct {
+	Type string
```
Comment (kristinaspring) [inline deviceAccess.go:56]: Can you add a comment here? The type of what? The type of...check?

File: deviceAccess.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	//Sep is the separator to be used to split the keys from the given paths
```
Comment (kristinaspring) [inline deviceAccess.go:58]:
```suggestion
	//Sep is the separator to be used to split the keys from the given paths.
```

File: primaryHandler.go [hunk 3]
```diff
@@ -51,99 +57,211 @@ const (
...
+	}
+}
+
+func GetLogger(ctx context.Context) bascule.Logger {
+	logger := log.With(logging.GetLogger(ctx), "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	return logger
+}
+
+//buildUserPassMap decodes base64-encoded strings of the form user:pass and write them to a map from user -> pass
+func buildUserPassMap(logger log.Logger, encodedBasicAuthKeys []string) (userPass map[string]string) {
```
Comment (kristinaspring) [inline primaryHandler.go:125]: I think we should put this in the `basculehttp` package.

Comment (kristinaspring) [reply primaryHandler.go:125]: Probably without the logger as a parameter...

Comment (joe94) [reply primaryHandler.go:126]: Yeah, that sounds like a good idea. I added this issue: https://github.com/xmidt-org/bascule/issues/57

File: main.go [hunk 4]
```diff
@@ -187,13 +189,16 @@ func talaria(arguments []string) int {
 	for exit := false; !exit; {
 		select {
 		case s := <-signals:
-			if s != os.Kill && s != os.Interrupt {
-				logger.Log(level.Key(), level.InfoValue(), logging.MessageKey(), "ignoring signal", "signal", s)
-			} else {
+			switch s {
+			case syscall.SIGURG:
+				// We are getting bombarded with SIGURGS due to Go1.14's new way to async preempt goroutines  https://github.com/golang/go/issues/37942
+				// Don't log as info as it will fill log with unnecessary entries
```
Comment (kristinaspring) [inline main.go:195]: We should make an issue in other service repos to fix this there as well.

File: primaryHandler.go [hunk 5]
```diff
@@ -180,23 +286,63 @@ func NewPrimaryHandler(logger log.Logger, manager device.Manager, v *viper.Viper
...
+	for _, check := range config.Checks {
+		parsedCheck, err := parseDeviceAccessCheck(check)
+		if err != nil {
+			errorLogger.Log(logging.ErrorKey(), err, logging.MessageKey(), "deviceAccesscheck parse failure")
+			return nil, errors.New("Failed parsing DeviceAccessChecks")
+		}
+		parsedChecks = append(parsedChecks, parsedCheck)
+	}
+
+	if config.Type == "enforce" || config.Type == "monitor" {
```
Comment (kristinaspring) [inline primaryHandler.go:335]: This code could be flatter by checking at the beginning of this function to verify that the type is "enforce" or "monitor", and returning an error if it's not either.  Then, the positive case will be the one returned at the end of this function. 🙂

Comment (joe94) [reply primaryHandler.go]: Good idea, thanks!

**Pull request merged**